src/main/flight/failsafe.c

```diff
@@ -248,22 +248,42 @@ void failsafeUpdateState(void)
                     failsafeState.phase = FAILSAFE_RX_LOSS_RECOVERED;
                 } else {
                     switch (failsafeConfig()->failsafe_procedure) {
-                        case FAILSAFE_PROCEDURE_AUTO_LANDING:
+                        if (IS_RC_MODE_ACTIVE(BOXAUTOLAND)) {
                             // Stabilize, and set Throttle to specified level
                             failsafeActivate();
                             break;
-
-                        case FAILSAFE_PROCEDURE_DROP_IT:
+                        }
+                        else if (IS_RC_MODE_ACTIVE(BOXDROP)) {
                             // Drop the craft
                             failsafeActivate();
                             failsafeState.phase = FAILSAFE_LANDED;      // skip auto-landing procedure
                             failsafeState.receivingRxDataPeriodPreset = PERIOD_OF_3_SECONDS; // require 3 seconds of valid rxData
                             break;
-                        case FAILSAFE_PROCEDURE_GPS_RESCUE:
+                        }
+                        else if (IS_RC_MODE_ACTIVE(BOXGPSRESCUE)) {
                             failsafeActivate();
                             failsafeState.phase = FAILSAFE_GPS_RESCUE;
                             failsafeState.receivingRxDataPeriodPreset = PERIOD_OF_3_SECONDS;
                             break;
+                        }
+                        else {
```

Comment:
This is completely butchering the `switch` statement. Please clean this up to put the `switch` either inside or outside the `if`, but not on both sides.